query-builder/src/test/java/com/datastax/oss/driver/api/querybuilder/relation/RelationTest.java

```diff
@@ -42,13 +44,78 @@ public void should_generate_is_not_null_relation() {
   }
 
   @Test
-  public void should_generate_in_relation() {
+  public void should_generate_contains_relation() {
+    assertThat(selectFrom("foo").all().where(Relation.column("k").contains(literal(1))))
+        .hasCql("SELECT * FROM foo WHERE k CONTAINS 1");
```

Comment:
A bit pedantic, so can choose to take these suggested changes or not, but it looks like `CONTAINS`/`CONTAINS KEY` and (`NOT`) variants, do not work on partition keys without `ALLOW FILTERING`, but can work with `ALLOW FILTERING` if you create an SAI index on a non-partition key column.   I think it would be good to make the tests model against valid queries as much as possible, so could we do something like this:

```suggestion
    assertThat(selectFrom("foo").all()
        .where(Relation.column("k").isEqualTo(literal(1)))
        .where(Relation.column("v").contains(literal(1))))
        .hasCql("SELECT * FROM foo WHERE k=1 AND v CONTAINS 1");
```

which would be valid for a schema like:

```sql
CREATE TABLE simple.foo (
    k int PRIMARY KEY,
    v set<int>
);

CREATE INDEX foo_sai_idx ON simple.foo (v) USING 'sai';
```

Although this doesn't work for `NOT CONTAINS`:

```
SELECT * FROM foo WHERE k=1 and v NOT CONTAINS 1;
InvalidRequest: Error from server: code=2200 [Invalid query] message="Cannot execute this query as it might involve data filtering and thus may have unpredictable performance. If you want to execute this query despite the performance unpredictability, use ALLOW FILTERING"
```

Which, based on the tests in C* and [this](https://issues.apache.org/jira/browse/CASSANDRA-18584?focusedCommentId=17877916&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17877916), it looks like this is by design which is surprising to me that there is no way to use an operator without `ALLOW FILTERING`?

Reply:
I see now in https://issues.apache.org/jira/browse/CASSANDRA-19930 that you provided a lot of context in comments around what is valid/not valid which now that I see it is very useful for my understanding 👍